### ISSUE TITLE
Crashes when file system is updated

### ISSUE THREAD

**Issue by rjplevin** [quicksilver/Quicksilver#2550]
QS quits several times a day, usually after downloading an item or making some other change to the file system. Looks like some kind of race condition or failure to respond properly to a file system change notification.

I'm running 10.15.7 but this has been happening for a very long time, like more than a year, and since at least one macOS upgrade.

**Comment by skurfer** [comment]
What version of Quicksilver are you running? There was a version that crashed every time it checked for updates (which prevented a lot of people from ever seeing those updates). The [current version][1] is 1.6.1.

[1]: https://qsapp.com/download.php

**Comment by rjplevin** [comment]
1.6.1 (4026)

**Comment by skurfer** [comment]
I can find the ones you submitted if you tell me the name of your computer (in System Preferences → Sharing).

**Comment by rjplevin** [comment]
It's "bebop".

**Comment by skurfer** [comment]
I finally got a chance to review those crash reports. They’re all for the same reason:

    Thread 5 Crashed:: Dispatch queue: QSCatalogEntry scanQueue: QSPresetDownloads

So any time something changes in your Downloads folder, Quicksilver rescans it and crashes. It’s probably from the same file each time.

If you’re willing, some detective work would help me out.

1. Move all of the files out of your Downloads folder to a temporary location.
2. Add them back one at a time until Quicksilver crashes. (It _should_ be picking up the changes as they happen in that folder, so there should be no need to rescan manually, wait 10 minutes, etc.)
3. Once you’ve identified the file, if it doesn’t contain anything private, could you share it with me so I can try to reproduce and fix the crash?

Thanks.

**Comment by rjplevin** [comment]
Thanks for looking into this. 

I have over 300 files in Downloads, so I moved them all out and moved them back with a bash loop, printing names, moving files back to ~/Downloads, and sleeping 1 sec between. As soon as QS crashed, I moved the dozen most recently moved files/dirs out of ~/Downloads again and reran the loop with a longer sleep. Unfortunately, it didn't crash again. 

I was thinking it might have to do a delay after moving a directory while QS processed the files within, so that the crash isn't immediate. After experimenting with some subdirectories, I was able to get QS to crash twice more, but I've failed to find anything repeatable. Anyway, I spent like an hour on this and haven't found the smoking gun. 

I'd be willing to run it with more verbose logging or in a debugger if that's possible.

-Rich
